File: programs-basic-java/src/test/java/com/local/numbertypes/PerfectNumberTest.java
package com.local.numbertypes;

import static com.local.numbertypes.NumberType.PERFECT_NUMBER;

import java.util.List;

import org.junit.jupiter.api.Test;

class PerfectNumberTest extends AbstractNumberTest {

	@Test
	void testPerfectNumbers() {
		testNumbersTypeByFile(PERFECT_NUMBER, "perfectnumbers.txt");
	}

	@Test
	void testPerfectNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(PERFECT_NUMBER, List.of(1,6,28,496), 4);
	}
}

File: programs-basic-kotlin/src/test/kotlin/com/local/numbertypes/PerfectNumberTest.kt
package com.local.numbertypes
import org.junit.jupiter.api.Test
class PerfectNumberTest : AbstractNumberTest() {
	@Test
	fun testPerfectNumbers() {
		testNumbersTypeByFile(com.local.numbertypes.NumberType.PERFECT_NUMBER, "perfectnumbers.txt")
	}

	@Test
	fun testPerfectNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(com.local.numbertypes.NumberType.PERFECT_NUMBER, java.util.List.of<Int>(1, 6, 28, 496), 4)
	}
}